packages/ses/src/tame-global-error-object.js

```diff
@@ -53,7 +96,51 @@ export default function tameGlobalErrorObject(errorTaming = 'safe') {
     },
   };
 
+  // A prepareFn is a prepareStackTrace function.
+  // An sst is a `structuredStackTrace`, which is an array of
+  // callsites.
+  // A user prepareFn is a prepareFn defined by a client of this API,
+  // and provided by assigning to `Error.prepareStackTrace`.
+  // A user prepareFn should only receive an attenuated sst, which
+  // is an array of attenuated callsites.
+  // A system prepareFn is the prepareFn created by this module to
+  // be installed on the real `Error` constructor, to receive
+  // an original sst, i.e., an array of unattenuated callsites.
+  // An input prepareFn is a function the user assigns to
+  // `Error.prepareStackTrace`, which might be a user prepareFn or
+  // a system prepareFn previously obtained by reading
+  // `Error.prepareStackTrace`.
+
+  // A weakset branding some functions as system prepareFns, all of which
+  // must be defined by this module, since they can receive an
+  // unattenuated sst.
+  const systemPrepareFnSet = new WeakSet();
+
+  const systemPrepareFnFor = inputPrepareFn => {
+    if (systemPrepareFnSet.has(inputPrepareFn)) {
+      return inputPrepareFn;
+    }
+    const systemMethods = {
```

Comment:
Does this object exist just to provide a safe `this` value for one of those tricky function-application modes? Clever. But subtle. I'd add a one-line comment to that effect so somebody like me isn't tempted to "make this more efficient" by removing the "unnused" wrapper object.

Reply:
Yes. It is so we can define the function using concise method syntax. This is the only way to define a function that
   * has no construct behavior or `.prototype` property (also true of arrow functions)
   * does not bind `this` (not true of arrow functions)